core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -186,13 +186,13 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     try {
       val statusList = Option(fs.listStatus(new Path(logDir))).map(_.toSeq)
         .getOrElse(Seq[FileStatus]())
-      var newLastModifiedTime = lastModifiedTime
       val logInfos: Seq[FileStatus] = statusList
         .filter { entry =>
           try {
             getModificationTime(entry).map { time =>
-              newLastModifiedTime = math.max(newLastModifiedTime, time)
-              time >= lastModifiedTime
+              val fileName = entry.getPath.getName
+              val oldLastModifiedTime = lastModifiedTimes.getOrElse(fileName, -1L)
+              time > oldLastModifiedTime
```

Comment:
I think technically this has to be `>=` since you might have an update that occurred in the same millisecond. This is very unlikely. But is there harm in reprocessing logs in this code? does it have to guarantee a log isn't read twice?